prov/efa/src/rxr/rxr_pkt_entry.c

```diff
@@ -201,6 +218,16 @@ struct rxr_pkt_entry *rxr_pkt_get_unexp(struct rxr_ep *ep,
 		unexp_pkt_entry = *pkt_entry_ptr;
 	}
 
+	/*
+	 * Increment the counter now to allocate and post another receive
+	 * buffer to SHM; changing the type prevents incrementing the counter
+	 * again when this unexpected message is matched and eventually freed.
+	 */
+	if (pkt_type == RXR_PKT_FROM_SHM_RX_POOL) {
```

Comment:
I think this bug applies to EFA rx packets too, when rxr_env.rx_copy_unexp is 0.

So we either get rid of rxr_env.rx_copy_unexp  (which I prefer), or do something similar for EFA (like increase counter) when rxr_env.rx_copy_unexp is 0.

Reply:
For sure, we should deprecate that option or at least put a big warning on it. I didn't want to make that change here since I think there are users of that option.

We can't do something similar for EFA since the RX pool is registered, the only option we have is to copy unexpected messages out of the registered RX pool.

Reply:
I see. If someone are using the option (setting FI_EFA_RX_COPY_UNEXP to 0), their application will likely to hang (for the same  reason shm cause hang). So I think we have to deprecate that option (and abort the application if someone is setting it). But I agree that can be in a separate PR.